### ISSUE TITLE
[navigation] Complex turn lanes.

### ISSUE THREAD

**Issue by vng** [organicmaps/organicmaps#4873]
https://www.openstreetmap.org/way/857509971

- Show complex turn lanes like:
![image](https://user-images.githubusercontent.com/175612/228211374-3f17c2a8-6862-4107-a624-b4ffe6dd75b1.png)

- No maneuver was calculated on this roundabout.

**Comment by vng** [comment]
https://www.openstreetmap.org/way/1124672390
https://www.openstreetmap.org/way/1080054997

![image](https://user-images.githubusercontent.com/175612/228214218-c0c94b2d-6482-4728-8842-b41ce36d050d.png)

Should show lanes _just before_ the correct maneuver? Confusing example here when driver can take Right lane for to early for unneeded maneuver.

**Comment by d4f5409d** [comment]
> > https://www.openstreetmap.org/way/1124672390
> > https://www.openstreetmap.org/way/1080054997
> > 
> > ![image](https://user-images.githubusercontent.com/175612/228214218-c0c94b2d-6482-4728-8842-b41ce36d050d.png)
> > 
> > Should show lanes _just before_ the correct maneuver? Confusing example here when a driver can take the Right lane too early for an unneeded maneuver.
> 
> Or we should have some kind of counter when it will happen. It's nice to prepare for turns.

Or maybe highlitght the turn arrow on the map and the turn lane icon on the guidance to a color that's the same so we can differentiate it

**Comment by d4f5409d** [comment]
We should rethink the UI in my opinion. With the part on the left it just looks like a bloat.

**Comment by aronkvh** [comment]
> > > https://www.openstreetmap.org/way/1124672390
> > > https://www.openstreetmap.org/way/1080054997
> > > ![image](https://user-images.githubusercontent.com/175612/228214218-c0c94b2d-6482-4728-8842-b41ce36d050d.png)
> > > Should show lanes _just before_ the correct maneuver? Confusing example here when a driver can take the Right lane too early for an unneeded maneuver.
> > 
> > 
> > Or we should have some kind of counter when it will happen. It's nice to prepare for turns.
> 
> Or maybe highlitght the turn arrow on the map and the turn lane icon on the guidance to a color that's the same so we can differentiate it
> 
> Like this: ![IMG_20230328_160601.png](https://user-images.githubusercontent.com/76624594/228264425-cd84c173-5484-4eb9-ae0b-30fcdd403abd.png)

That wouldn't make sense to me.
Lane guidance means," this thing appeared so I should switch to this lane now". Having basically the same but with more arrows isn't really helpful.

**Comment by Misalf-git** [comment]
Quoteception makes me scroll a lot!

**Comment by nvarner** [comment]
#4886 may resolve this

**Comment by matheusgomesms** [comment]
Did #4907 solved these problems? If yes, this could be closed.

**Comment by zyphlar** [comment]
@matheusgomesms I don't think so (issue persists, the fundamental issue here is that the lane indicator is for *when you need to turn* and not *right now with the current lane situation.* In the example screenshot, get in the right turn lane, but *only when the turn indicator reads 50m, not now.* Which is confusing and not how other apps work.

I do think that #5101 would help with this, however. The lane indicator would always be present (or be present any time a lane choice matters) and only represent the final turning lane arrangement when it's actually time to prepare for the turn. The fix for one might be the fix for the other.

As for bullet point 1 of this two-part issue, here is another use case: https://www.openstreetmap.org/way/31804805 when turning right (westbound to northbound) the lane indicator displays a "go straight in lane 1 or turn right in lane 2" icon, however that's not accurate. The lane tagging and direction should be "use either lane 1 (straight/turn, prefer turn) or lane 2 (turn only) to turn right." Maybe the straight portion of lane 1 could be gray and only the right turn portion of lane 1 and lane 2 could be white.

**Comment by soshial** [comment]
Any updates on this issue with complex turn indicators (arrows)?

**Comment by AndrewShkrob** [comment]
There is a WIP PR with lanes refactoring but a lot of work has to be done and I have no desire right now to continue my work
https://github.com/organicmaps/organicmaps/pull/7077
